packages/react-core/src/components/Banner/__tests__/Banner.test.tsx

```diff
@@ -1,27 +1,87 @@
-  test(`${variant} banner`, () => {
-    const { asFragment } = render(
-      <Banner
-        variant={variant as 'default' | 'info' | 'success' | 'warning' | 'danger'}
-        aria-label={variant}
-        screenReaderText={`${variant} banner`}
-      >
-        {variant} Banner
-      </Banner>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
-});
-
-test(`sticky banner`, () => {
-  const { asFragment } = render(
-    <Banner aria-label="sticky" isSticky>
-      Sticky Banner
-    </Banner>
+
+test('Renders without children', () => {
+  render(
+    <div data-testid="banner">
+      <Banner />
+    </div>
   );
+  expect(screen.getByTestId('banner').firstChild).toBeVisible();
+});
+
+test('Renders children', () => {
+  render(<Banner>Test</Banner>);
+  expect(screen.getByText('Test')).toBeVisible();
+});
+
+test('Renders with class name pf-c-banner', () => {
+  render(<Banner>Test</Banner>);
+  expect(screen.getByText('Test')).toHaveClass('pf-c-banner');
+});
+
+test('Renders with custom class name when className prop is provided', () => {
+  render(<Banner className="custom-class">Test</Banner>);
+  expect(screen.getByText('Test')).toHaveClass('custom-class');
+});
+
+test('Renders without any modifier class when variant prop is not passed', () => {
+  render(<Banner>Test</Banner>);
+  expect(screen.getByText('Test')).toHaveClass('pf-c-banner', { exact: true });
+});
+
+test('Renders with class name pf-m-success when "success" is passed to variant prop', () => {
+  render(<Banner variant="success">Test</Banner>);
+  expect(screen.getByText('Test')).toHaveClass('pf-m-success');
+});
+
+test('Renders with class name pf-m-danger when "danger" is passed to variant prop', () => {
+  render(<Banner variant="danger">Test</Banner>);
+  expect(screen.getByText('Test')).toHaveClass('pf-m-danger');
+});
+
+test('Renders with class name pf-m-warning when "warning" is passed to variant prop', () => {
+  render(<Banner variant="warning">Test</Banner>);
+  expect(screen.getByText('Test')).toHaveClass('pf-m-warning');
+});
+
+test('Renders with class name pf-m-info when "info" is passed to variant prop', () => {
+  render(<Banner variant="info">Test</Banner>);
+  expect(screen.getByText('Test')).toHaveClass('pf-m-info');
+});
+
+test('Renders screenReaderText as "default banner" by default', () => {
+  render(<Banner>Test</Banner>);
+  expect(screen.getByText('default banner')).toHaveTextContent(/^default banner$/);
+});
+
+test('Renders screenReaderText as "success banner" when variant="success"', () => {
+  render(<Banner variant="success">Test</Banner>);
+  expect(screen.getByText('success banner')).toHaveTextContent(/^success banner$/);
+});
+
+test('Renders custom screenReaderText passed via prop', () => {
+  render(<Banner screenReaderText="Custom screen reader text">Test</Banner>);
+  expect(screen.getByText('Custom screen reader text')).toHaveTextContent(/^Custom screen reader text$/);
+});
+
```

Comment:
Also can you add a test to make sure the `screenReaderText` has the `pf-u-screen-reader` class? Missed that on my first review.